maven.gradle

```diff
@@ -0,0 +1,27 @@
+def properties = new Properties()
+def propsFile = file("$rootDir/publish.properties")
+if (propsFile.exists()) {
+    properties.load(propsFile.newInputStream())
+}
+
+/*
+* 1. findProperty("sonatypeUsername"): use parameter from the command (if attached) so you can run ->
+* /gradlew deploy -PsonatypeUsername={yourMavenUsername} -PsonatypePassword={yourMavenPassword}.
+* 2. properties["sonatype.user"]: use property defined in publish.properties file in the root directory. If you have that file in the root
+* of the project you can run -> ./gradlew deploy
+* 3. System.getenv("SONATYPE_PASS"): use global variable defined in the system. No need to create publish.properties file, you can simply
+* run -> ./gradlew deploy.
+*/
+ext.sonatype = [
+
+        url     : properties["sonatype.url"]
+                ?: "https://oss.sonatype.org/service/local/staging/deploy/maven2",
+
+        username: findProperty("sonatypeUsername")?.toString()
+                ?: properties["sonatype.user"]
+                ?: System.getenv("SONATYPE_USERNAME"),
```

Comment:
Btw, I have these env variables defined and I avoid using publish.properties, because it works on all projects automatically, and don't have to think about updating gitignore

Reply:
Yes, I would suggest removing this line from every library and the template if it exists there, so we don't encourage  using a file that can be easily pushed to gitignore

Reply:
👌 
I will recheck all libs, in library template there is no such file

Reply:
I'm sorry, I didn't mean to check if there is a file. I suggested removing the usage of the properties file in this file.